RiotSwiftUI/Modules/LocationSharing/StartLocationSharing/Coordinator/LocationSharingCoordinator.swift

```diff
@@ -166,17 +166,22 @@ final class LocationSharingCoordinator: Coordinator, Presentable {
     
     // Check if user can send beacon info state event
     private func canShareLiveLocation() -> Bool {
-        guard let myUserId = parameters.roomDataSource.mxSession.myUserId else {
+        guard let myUserId = parameters.roomDataSource.mxSession.myUserId,
+              let roomPowerLevels = parameters.roomDataSource.roomState.powerLevels,
+              let userPowerLevel = RoomPowerLevel(rawValue: parameters.roomDataSource.roomState.powerLevels.powerLevelOfUser(withUserID: myUserId)) else {
             return false
         }
         
-        let userPowerLevelRawValue = parameters.roomDataSource.roomState.powerLevels.powerLevelOfUser(withUserID: myUserId)
-        
-        guard let userPowerLevel = RoomPowerLevel(rawValue: userPowerLevelRawValue) else {
-            return false
-        }
-        
-        return userPowerLevel.rawValue >= RoomPowerLevel.moderator.rawValue
+        // Should call `minimumPowerLevelForSendingStateEvent(_ eventType: MXEventType) -> Int` from `MatrixSDK:MXRoomPowerLevels.swift`
+        // but can't because it is inaccessible due to 'internal' protection level
```

Comment:
That seems purely like an oversight when refining the objc for Swift. Would you mind making these 2 public and updating the PR based upon that:

https://github.com/matrix-org/matrix-ios-sdk/blob/develop/MatrixSDK/Contrib/Swift/JSONModels/MXRoomPowerLevels.swift

It would give us a much neater implementation here.

Reply:
Done.

Need to compile against updated matrix-iso-sdk lib (https://github.com/matrix-org/matrix-ios-sdk/pull/1869)

Reply:
Perfect, it's merged so feel free to update the submodule and push 👍

Reply:
Sorry, what do you mean by 'update the submodule and push" ?

Reply:
Ah, the SDK as a pod was deprecated recently, and EI switched to consuming it as a [git submodule](https://github.com/element-hq/element-ios/pull/7804). Don't worry, I've opened https://github.com/element-hq/element-ios/pull/7819 with the update, so once that is merged just rebase on `develop` and CI should be happy.

Reply:
Ok, thank you @pixlwave.

I followed the matrix-sdk pod deprecation, but didn't follow how to integrate matrix-sdk update into git-submodule.

I will study #7819 👍